.github/workflows/sycl_detect_changes.yml

```diff
@@ -0,0 +1,48 @@
+name: Determine which check-* targets could be affected by the PR
```

Comment:
Let's use something short. AFAIK, this name appears in places where only first N symbols are displayed. For example https://github.com/intel/llvm/actions (see left side). Another example - https://github.com/intel/llvm/actions/runs/5193089878.

How about "Identify impacted LIT tests"?

Do we have to name the workflow? I see that `need_check` name is very similar to this.